projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/class-wpcom-admin-bar.php

```diff
@@ -60,6 +60,17 @@ public function add_node( $args ) {
 			return;
 		}
 
+		if ( $args['id'] === 'my-account' ) {
+			if ( ! is_user_member_of_blog() || get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
```

Comment:
I extracted the logic from here: https://github.com/Automattic/jetpack/blob/9b6a4f53acd2b8a89ec0abb1b74234d515b3bf38/projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php#L176 since it should be the same.